### PR TITLE
DS-1349: Add configuration to hide submitter in item history (JSPUI).

### PR DESCRIPTION
**Pull request by pnbecker**
https://jira.duraspace.org/browse/DS-1349

### CODE REVIEW

**Comment by pnbecker** [comment]
This PR is currently untested. I developed and tested it before the Service Based API came in. As versioning is currently broken, I was unable to test my rebase yet.

**Comment by pnbecker** [comment]
This PR contains DSPR#1089 as it was impossible to create this PR without it. As soon as DSPR#1089 I'm going to rebase this again. Everything else is done, PR is ready for review again.

**Comment by pnbecker** [comment]
Rebased to resolve merge conflict.

File: dspace-jspui/src/main/java/org/dspace/app/webui/servlet/VersionHistoryServlet.java [hunk 1]
```diff
@@ -62,6 +63,11 @@ protected void doDSGet(Context context, HttpServletRequest request,
         	throw new IllegalArgumentException("Item is null");
         }
         
+        Boolean show_submitter = new DSpace()
+                .getConfigurationService()
+                .getPropertyAsType("versioning.item.history.include.submitter",
+                        new Boolean(false));
```
Comment (mwoodiupui) [inline dspace-jspui/src/main/java/org/dspace/app/webui/servlet/VersionHistoryServlet.java]: Boolean.FALSE

File: dspace-jspui/src/main/java/org/dspace/app/webui/servlet/VersionHistoryServlet.java [hunk 2]
```diff
@@ -71,9 +77,12 @@ protected void doDSGet(Context context, HttpServletRequest request,
             {
                 throw new AuthorizeException();
             }
-
+        } else {
+            // if user is Admin override show_submitter
+            show_submitter = new Boolean(true);
```
Comment (mwoodiupui) [inline dspace-jspui/src/main/java/org/dspace/app/webui/servlet/VersionHistoryServlet.java]: Boolean.TRUE

**Comment by tdonohue** [comment]
While this looks like it'd "work" for JSPUI, it has the feel of something that we might think about pushing down lower in the API "stack".  Ideally, if this config was disabled, the submitter info would not even be *provided* to the UI layer. This would ensure that it didn't get displayed to JSPUI or XMLUI or even REST (or other integration points).  

Fixing it at the JSPUI is nice. But, ideally we'd find a way to resolve this lower in the stack.

**Comment by pnbecker** [comment]
@tdonohue while I see your point, we need this information for logged in administrators, so we have to pass it to the UI anyway.

**Comment by helix84** [comment]
@pnbecker I don't think that changes anything about the concern - your authorization should be recognized below the API layer.

**Comment by pnbecker** [comment]
@helix84 The problem is, that your request requires a complete change of how the version history table in JSPUI works. This would require a massive change. Also would it result in a version history table that works completely different compared to other parts of the UI regarding the point how data is served.

We have a concrete issue. In the past we resolved issues of that kind in the UI. That can be considered being bad design, but that is one of the reasons we're currently discussing a complete change of the UI. For now, I don't think that it is reasonable to change this here.

**Comment by mwoodiupui** [comment]
I agree that we really should be making access decisions much further down the stack, but I also agree that the necessary changes would be much too far-ranging and disruptive to be done this close to release.  I reluctantly support a strategy of "make it work now, and make it right later."

What I would like to see, some day, is that **all** access decisions are made by the content objects or their services.  Callers would receive refusal of a request for access to a single forbidden item, and on aggregate requests would receive only the items permitted them.  This requires that objects and their services be told who is asking, and for what purpose.  The former information is carried in the Context.  How we will convey the latter, I do not yet know.  It will require a large expansion of the set of access request types.

File: dspace-jspui/src/main/java/org/dspace/app/webui/servlet/VersionHistoryServlet.java (hunk 1 above)
Comment (tdonohue) [inline dspace-jspui/src/main/java/org/dspace/app/webui/servlet/VersionHistoryServlet.java]: This should just use `getBooleanProperty()` method of the ConfigurationService. Not sure why it needs to use `getPropertyAsType`?

**Comment by tdonohue** [comment]
I'm OK with letting this go in for now. Ideally, longer term, we need to find a way to do this at a lower level, so that the flag can be turned on/off for all access points (UIs or REST or whatever) at once.

**Comment by pnbecker** [comment]
Merging this as discussed in today's developer meeting.

**Pull request merged**